test/validations-test.js

```diff
@@ -1,73 +1,66 @@
+      const expression = `
+      {% computeColumn testTable test_column %}
+      {% if x %}
+      ...
+      {% elsif y %}
+      ...
+      {% else %}
+      ...
+      {% endif %}
+      {% for item in items %}
+      ...
+      {% endfor %}
+      {% assign $$answer = test %}
+      {% endcomputeColumn %}
+    `;
 
-    const validationErrors = engine.checkValidJSON(expression);
-    expect(validationErrors).to.deep.equal([]);
-  });
+      const expectedValidationErrors = [];
+      const validationErrors = checkValidDynamicTable(engine, expression);
+      expect(validationErrors).to.deep.equal(expectedValidationErrors);
+    });
 
-  it("should return array of errors if parseAssign is assigned invalid JSON value", () => {
-    const engine = commonSetup();
+    it("should return an array of errors if there are no occurences of $$answer = something which is not in any if, for or unless block", () => {
+      const expression = `
+      {% computeColumn testTable test_column %}
+      {% if x %}
+      ...
+      {% elsif y %}
+      ...
+      {% else %}
+      ...
+      {% endif %}
+      {% for item in items %}
+      ...
+      {% endfor %}
+      {% endcomputeColumn %}
+    `;
 
-    const expression = `
-        {% parseAssign x0 = 10, 0 %}
-        {% parseAssign y = a | plus: b %}
-        {% if a == "USD" and b == "INR" % }
-          {% parseAssign x1 = {"key": value } %}  <!-- Invalid JSON -->
-        {% elsif a == "INR" %}
-          {% parseAssign x2 = [1, 2,] %}      <!-- Invalid JSON -->
-        {% else %}
-          {% parseAssign x3 = unquotedString %} <!-- Invalid JSON -->
-        {% endif %}
-      `;
-    const expectedValidationErrors = [
-      {
-        expression: "parseAssign x0 =  10, 0",
-        errorMessage:
-          "Invalid value assigned to parseAssign statement at line 2",
-      },
-      {
-        expression: "parseAssign y =  a | plus: b",
-        errorMessage:
-          "Invalid value assigned to parseAssign statement at line 3",
-      },
-      {
-        expression: "parseAssign x2 =  [1, 2,]",
-        errorMessage:
-          "Invalid value assigned to parseAssign statement at line 7",
-      },
-      {
-        expression: "parseAssign x3 =  unquotedString",
-        errorMessage:
-          "Invalid value assigned to parseAssign statement at line 9",
-      },
-    ];
-    const validationErrors = engine.checkValidJSON(expression);
-    expect(validationErrors).to.deep.equal(expectedValidationErrors);
+      const expectedValidationErrors = [
+        {
+          message:
+            "$$answer is not assigned outside any loops or condition block",
+          metadata: {
+            expression:
+              "\n      {% computeColumn testTable test_column %}\n      {% if x %}\n      ...\n      {% elsif y %}\n      ...\n      {% else %}\n      ...\n      {% endif %}\n      {% for item in items %}\n      ...\n      {% endfor %}\n      {% endcomputeColumn %}\n    ",
```

Comment:
I don't think we need to print the whole computation...it can be quite long. Only tableName and columnName is enough